internal/infrastructure/kubernetes/infra_resource.go

```diff
@@ -73,6 +73,14 @@ func (i *Infra) createOrUpdateDeployment(ctx context.Context, r ResourceRender)
 	}
 
 	return i.Client.CreateOrUpdate(ctx, key, current, deployment, func() bool {
+		// applied to k8s the "DeprecatedServiceAccount" will fill it.
+		deployment.Spec.Template.Spec.DeprecatedServiceAccount = current.Spec.Template.Spec.DeprecatedServiceAccount
+		// applied to k8s the "SecurityContext" will fill it with default settings.
```

Comment:
aren't we just overriding security context here, so the user defined values will be lost ?